docs/source/opener_parameters_api.md

```diff
@@ -0,0 +1,220 @@
+full and detailed enough to allow the automatic construction of a user
+interface for access to the available datasets. Note that, under this system:
+
+  1. Every dataset which an opener provides can support a different set of
+     opener parameters.
+
+  2. The schema does not allow the representation of interdependencies
+     between values of opener parameters within a dataset. For
+     instance, the following cannot be represented in an open
+     parameters schema:
+     
+     *sensor_type*: A or B  
+     *variables*: [temperature, humidity] for sensor type A;
+                  [temperature, pressure] for sensor type B
+
+To work around some of the restrictions of point (2) above, a dataset MAY be
+presented by the opener as multiple "virtual" datasets with difference
+parameter schemas. For instance, the hypothetical dataset described above
+could be offered not as a single dataset `envdata` but as two datasets
+`envdata:sensor-a` and `envdata:sensor-b`, with different sets of permitted
+variables.
+
+Sometimes, the interdependencies between parameters are too complex to
+be represented fully by splitting datasets in this way. In these cases:
+
+  1. The JSON Schema SHOULD describe the smallest possible superset of the
+     allowed parameter combinations.
+
+  2. The additional restrictions on parameter combinations MUST be clearly
+     documented.
+
+  3. If illegal parameter combinations are supplied, the opener MUST raise an
+     exception with an informative error message, and the user interface
+     should present this message clearly to the user.
+
+## Common parameters
+
+While an opener is free to define any opener parameters for any of its
+datasets, there are some common parameters which are likely to be used by the
+majority of datasets. Any dataset requiring the specification of these
+parameters MUST use the standard parameter names, syntax, and semantics
+defined below, in order to keep the interface consistent. For instance, if a
+dataset allows a time aggregation period to be specified, it MUST use the
+`time_period` parameter with the format described below rather than some other
+alternative name and/or format. The parameters are described below with their
+Python type annotations.
+
+ - `variable_names: List[str]`  
+   A list of the identifiers of the requested variables.
+ - `bbox: Union[str,Tuple[float, float, float, float]]`
+   The bounding box for the requested data, in the order x0, y0, x1, y1.
+ - `crs: str`  
+   The identifier for the co-ordinate reference system of geographic data.
+ - `spatial_res: float`  
+   The requested spatial resolution (x and y) of the returned data.
+ - `time_range: Tuple[Optional[str], Optional[str]]`  
+   The requested time range for the data to be returned. See section
+   ‘Date, time, and duration specifications’ below.
+ - `time_period: str`  
+   The requested temporal aggregation period for the data. See section
+   ‘Date, time, and duration specifications’ below.
```

Comment:
Is it possible to add in-page ref for _‘Date, time, and duration specifications’_?

Reply:
Done.